Starting screen menu

FrederikSchlemmer opened the pull request:
Fixes #29

Added a menu to the starting screen. Interaction with the mouse in the menu, but still the player control with the keyboard.

FrederikSchlemmer committed:
Starting screen menu

Added a menu to the starting screen. Interaction with the mouse in the menu, but still the player control with the keyboard.

FrederikSchlemmer committed:
Update BrainySnake.java

Added try / catch for loading the file

RudiKlassen-zz commented on core/src/de/adesso/brainysnake/BrainySnake.java:
```diff
@@ -102,22 +126,67 @@ private void initializeCamera() {
+        stage.addActor(newGameButton);
+
+        // Settings Button for version 1.1
+//        TextButton settingsButton = new TextButton("Settings", skin);
```
Pls never commit commented out code

FrederikSchlemmer replied:
Deleted the commented out code.

RudiKlassen-zz commented on core/src/de/adesso/brainysnake/BrainySnake.java:
```diff
@@ -189,9 +258,14 @@ public void drawGameOverScreen() {
+            texture.dispose();
+            gameSpriteBatch.dispose();
+            pixmap.dispose();
+        } catch (Exception e){
```
this is not necessary. That would happen anyway in the case of an exception

FrederikSchlemmer replied:
Deleted this try/catch block.

RudiKlassen-zz commented on core/src/de/adesso/brainysnake/BrainySnake.java:
```diff
@@ -44,6 +59,18 @@
     @Override
     public void create() {
         pixmap = new Pixmap(WIDTH, HEIGHT, Pixmap.Format.RGBA8888);
+        stage = new Stage();
```
für bessere lesbarkeit, bitte sprechende namen verwenden like _pauseStage_ or _gameplayStage_ (In IntelliJ you can refactor names with ALT+F6)

FrederikSchlemmer replied:
Renamed the variable to mainStage. The same stage will be used in the other screens.

RudiKlassen-zz commented on core/src/de/adesso/brainysnake/BrainySnake.java:
```diff
@@ -44,6 +59,18 @@
+        createBasicSkin();
+        
+        try {
+            logoTexture = new Texture(Gdx.files.internal("./core/src/de/adesso/brainysnake/img/BrainySnake_Headline.png"));
```
The header Image png should be moved to core assets folder (it´s a common asset). Then you can load it with _new Texture("[path within assets folder]BrainySnake_Headline.png");_

FrederikSchlemmer replied:
Moved the image to the the assets folder and changed the loading.

RudiKlassen-zz commented on core/src/de/adesso/brainysnake/BrainySnake.java:
```diff
@@ -44,6 +59,18 @@
+            TextureRegion region = new TextureRegion(logoTexture, 0, 0, 629, 180);
+            logoBrainySnake = new Image(region);
+            logoBrainySnake.setPosition(Config.APPLICATION_WIDTH / 4, Config.APPLICATION_HEIGHT - 250);
+        } catch (Exception e){
```
this is not necessary, because in the exception case, that would happen anyway

FrederikSchlemmer replied:
Deleted this try/catch block.

RudiKlassen-zz commented on core/src/de/adesso/brainysnake/BrainySnake.java:
```diff
@@ -56,7 +83,8 @@ public void create() {
         game = new Game();
         game.init(HEIGHT, WIDTH);
-        Gdx.input.setInputProcessor(new KeyBoardControl(game));
+
```
Good bye KeyboardControl for KeyBoardControl-Snake ;) Pls change the Input Processor within gameplay to keyboard.

FrederikSchlemmer replied:
Usage of InputMultiplexer to add multiple InputProcessors.

RudiKlassen-zz replied:
Top

FrederikSchlemmer committed:
Fixed the requested changes

Deleted commented out code.
Deleted try/catch blocks.
Renamed the stage variable.
Readded the keyboard control.

FrederikSchlemmer committed:
Added multiple InputProcessor

Usage of InputMultiplexer

FlorianTim commented on core/src/de/adesso/brainysnake/BrainySnake.java:
```diff
@@ -59,17 +59,13 @@
-            LOGGER.error(e.toString());
-        }
+
+        logoTexture = new Texture("./core/assets/BrainySnake_Headline.png");
```
Move the png to the global assets folder
https://github.com/libgdx/libgdx/wiki/Managing-your-assets

FrederikSchlemmer replied:
Done.

FlorianTim replied:
Meinte den übergeordneten assets Ordner. Wie beim Logo

FrederikSchlemmer replied:
Ich verstehe das gerade nicht 100%. Wurde das nicht in dem Commit (54051b5550dee6dc4933722504f37fd99fe4ac39) erledigt?

RudiKlassen-zz replied:
@FrederikSchlemmer Jup. Das ist erledigt. Der Change oben ist veraltet.

FrederikSchlemmer committed:
Changed path to assets folder

RudiKl committed:
Merge branch 'master' into starting-screen-menu

RudiKl committed:
Fix dobble dispose of gameSpriteBatch
